### DISCUSSION TITLE
Variables for cache paths?

### DISCUSSION THREAD

**Discussion by transcendr** [TheR1D/shell_gpt#123] [Q&A] [answered]
I'd like to use environment variable to define the cache directory paths.  Setting the config options in .sgptrc works however the paths are static and so I can't share a conversation cache between two accounts on my mac (by setting the cache path to a shared Google Drive volume).  Unfortunately the path to the shared drive is different for each user account.  

1. Any suggested workarounds?
2. Any possibility of adding the option to get those paths from environment variables rather than the rc file?

Thanks, sgpt is awesome!

**Comment by TheR1D** [comment, marked as answer]
Thank you for the feedback! It is possible to use environment variables for any parameters defined in the [rc file](https://github.com/TheR1D/shell_gpt#runtime-configuration-file). You can use the same names. For instance, you can define:
```shell
export CHAT_CACHE_PATH=/path/to/cache
```
ShellGPT prioritizes ENV variables. In this case, no matter where CHAT_CACHE_PATH in the rc file points, the one from ENV will be used.

**Reply by transcendr** [reply]
Hey appreciate your quick response.  I did try that initiall, but it didn't seem to pick up my env variables (zsh shell) by itself.  However, I find that it does work if I set up an alias like this:

```
CHAT_CACHE_PATH="$RSHARED/shell_gpt/chat_cache"
CACHE_PATH="$RSHARED/shell_gpt/cache"

alias sg='CHAT_CACHE_PATH=$CHAT_CACHE_PATH CACHE_PATH=$CACHE_PATH sgpt'
```

Thanks again for the additional info!

**Reply by transcendr** [reply]
BTW, just to put a finer point on it, the `OPENAI_API_KEY` env variable is working without having to prefix the alias with it, but the cache paths do not seem to register.

**Reply by TheR1D** [reply]
I think you need to export the environment variable as I suggested before, so that child processes can access it. I tested it in my `zsh` shell and it works as expected.

**Reply by transcendr** [reply]
Ah yes, that's correct. Stupid oversight. Thanks again.

**Comment by edshamis** [comment]
@TheR1D I put this in sgptrc
```
CACHE_PATH="$SHELL_GPT_CACHE/cache"
```
With this in zshrc
```
export SHELL_GPT_CACHE=~/.cache/shell_gpt
```
but it doesn't use env var instead creates literally ~/"$SHELL_GPT_CACHE/ dir

**Reply by transcendr** [reply]
Dot files don't generally parse variables.  Try simply exporting them from .zshrc and they will take priority over the ones in a .sgptrc file if you need nonstatic paths.

```bash
# SGPT
# =================================================
# API KEY
export OPENAI_API_KEY=<your key here>
export OPENAI_API_HOST=https://api.openai.com
# Max amount of cached message per chat session.
export CHAT_CACHE_LENGTH=100
# Chat cache folder.
export CHAT_CACHE_PATH="$RSHARED/shell_gpt/chat_cache"
# Request cache length (amount).
export CACHE_LENGTH=100
# Request cache folder.
export CACHE_PATH="$RSHARED/shell_gpt/cache"
# Request timeout in seconds.
export REQUEST_TIMEOUT=60
```
Note: please check the README as these may have changed since time of this writing.

**Reply by edshamis** [reply]
I personally think dot files generally should parse variables. But it's just my opinion. 
Another problem with these variables (REQUEST_TIMEOUT, CACHE_PATH) that they're too general for exporting. If we want them to be exportable maybe better perpend them with for example SGPT_
